packages/core/src/fhir-to-cda/cda-templates/components/allergies.ts

```diff
@@ -0,0 +1,239 @@
+import { AllergyIntolerance, Bundle, Coding } from "@medplum/fhirtypes";
+import { isAllergyIntolerance } from "../../../external/fhir/shared";
+import { AllergiesSection } from "../../cda-types/sections";
+import {
+  ConcernActEntry,
+  ObservationEntryRelationship,
+  ObservationTableRow,
+} from "../../cda-types/shared-types";
+import {
+  buildCodeCe,
+  buildInstanceIdentifier,
+  buildValueCd,
+  formatDateToCdaTimestamp,
+  formatDateToHumanReadableFormat,
+  getTextFromCode,
+  isLoinc,
+  withoutNullFlavorObject,
+} from "../commons";
+import {
+  NOT_SPECIFIED,
+  extensionValue2014,
+  extensionValue2015,
+  loincCodeSystem,
+  loincSystemName,
+  oids,
+  placeholderOrgOid,
+} from "../constants";
+import { createTableRowsAndEntries } from "../create-table-rows-and-entries";
+import { initiateSectionTable } from "../table";
+import { AugmentedAllergy } from "./augmented-resources";
+
+export const allergiesSectionName = "allergies";
+const tableHeaders = [
+  "Substance",
+  "Category",
+  "Reaction",
+  "Date of Occurrence",
+  "Date of First Onset",
+  "Comments",
+];
+
+export function buildAllergies(fhirBundle: Bundle): AllergiesSection {
+  const allergies: AllergyIntolerance[] =
+    fhirBundle.entry?.flatMap(entry =>
+      isAllergyIntolerance(entry.resource) ? [entry.resource] : []
+    ) || [];
+
+  if (allergies.length === 0) {
+    return undefined;
+  }
+
+  const augmentedAllergies = allergies.map(allergy => {
+    return new AugmentedAllergy(allergiesSectionName, allergy);
+  });
+
+  const { trs, entries } = createTableRowsAndEntries(
+    augmentedAllergies,
+    createTableRowFromAllergyIntolerance,
+    createEntryFromAllergy
+  );
+
+  const table = initiateSectionTable(allergiesSectionName, tableHeaders, trs);
+
+  const allergySection = {
+    templateId: buildInstanceIdentifier({
+      root: oids.allergiesSection,
+      extension: extensionValue2015,
+    }),
+    code: buildCodeCe({
+      code: "48765-2",
+      codeSystem: loincCodeSystem,
+      codeSystemName: loincSystemName,
+      displayName: "Allergies and Adverse Reactions",
+    }),
+    title: "ALLERGIES, ADVERSE REACTIONS, ALERTS",
+    text: table,
+    entry: entries,
+  };
+  return allergySection;
+}
+
+function createTableRowFromAllergyIntolerance(
```

Comment:
can we use string joins or something else instead of hardcoding 0 indexing in this function?

Reply:
kinda like the way you use getDisplaysFromCodeableConcepts in encounters

Reply:
💯